--- a/sysdeps/unix/sysv/linux/riscv/sys/ucontext.h	2018-01-22 17:48:32.000000000 +0100
+++ b/sysdeps/unix/sysv/linux/riscv/sys/ucontext.h	2018-10-05 21:52:01.150006920 +0200
@@ -49,11 +49,11 @@
 /* Container for floating-point state.  */
 typedef union __riscv_fp_state fpregset_t;
 
+#endif
+
 /* Context to describe whole processor state.  */
 typedef struct sigcontext mcontext_t;
 
-#endif
-
 /* Userlevel context.  */
 typedef struct ucontext
   {
