--- /tmp/gpm-1.20.7/src/mice.c	2012-10-26 23:21:38.000000000 +0200
+++ gpm-1.20.7/src/mice.c	2017-07-15 16:13:35.546468279 +0200
@@ -238,19 +238,35 @@
 static int M_evdev (Gpm_Event * state, unsigned char *data)
 {
    struct input_event thisevent;
+   int b;
+   
    (void) memcpy (&thisevent, data, sizeof (struct input_event));
    if (thisevent.type == EV_REL) {
-      if (thisevent.code == REL_X)
-         state->dx = (signed char) thisevent.value;
-      else if (thisevent.code == REL_Y)
-         state->dy = (signed char) thisevent.value;
+       if (thisevent.code == REL_X) {
+           state->dx = (signed char) thisevent.value;
+       } else if (thisevent.code == REL_Y) {
+           state->dy = (signed char) thisevent.value;
+       } else if (thisevent.code == REL_WHEEL) {
+           state->wdy = thisevent.value;
+       }
+   } else if (thisevent.type == EV_ABS) {
+      /* XXX: we hardcode the range to 32768 */
+      if (thisevent.code == ABS_X)
+          state->x = ((thisevent.value * win.ws_col) / 32768) + 1;
+      else if (thisevent.code == ABS_Y)
+          state->y = ((thisevent.value * win.ws_row) / 32768) + 1;
    } else if (thisevent.type == EV_KEY) {
       switch(thisevent.code) {
-         case BTN_LEFT:    state->buttons ^= GPM_B_LEFT;    break;
-         case BTN_MIDDLE:  state->buttons ^= GPM_B_MIDDLE;  break;
-         case BTN_RIGHT:   state->buttons ^= GPM_B_RIGHT;   break;
-         case BTN_SIDE:    state->buttons ^= GPM_B_MIDDLE;  break;
+      case BTN_LEFT:    b = GPM_B_LEFT;    break;
+      case BTN_MIDDLE:  b = GPM_B_MIDDLE;  break;
+      case BTN_RIGHT:   b = GPM_B_RIGHT;   break;
+      case BTN_SIDE:    b = GPM_B_MIDDLE;  break;
+      default: b = 0; break;
       }   
+      if (thisevent.value)
+          state->buttons |= b;
+      else
+          state->buttons &= ~b;
    }
    return 0;
 }
@@ -2401,9 +2417,12 @@
            "", M_etouch, I_etouch, STD_FLG,
                                 {0xFF, 0x55, 0xFF, 0x54}, 7, 1, 0, 1, NULL}, 
 #ifdef HAVE_LINUX_INPUT_H
-   {"evdev", "Linux Event Device",
+   {"evdev", "Linux Event Device (mouse)",
             "", M_evdev, I_empty, STD_FLG,
                         {0x00, 0x00, 0x00, 0x00} , 16, 16, 0, 0, NULL},
+   {"evdev_abs", "Linux Event Device (tablet)",
+            "", M_evdev, I_empty, STD_FLG,
+                        {0x00, 0x00, 0x00, 0x00} , 16, 16, 0, 1, NULL},
 #endif /* HAVE_LINUX_INPUT_H */
    {"exps2",   "IntelliMouse Explorer (ps2) - 3 buttons, wheel unused",
            "ExplorerPS/2", M_imps2, I_exps2, STD_FLG,
